test/integration/util/util.go

```diff
@@ -131,3 +146,326 @@ func StartFakePVController(clientSet clientset.Interface) ShutdownFunc {
 	informerFactory.Start(ctx.Done())
 	return ShutdownFunc(cancel)
 }
+
+// TestContext store necessary context info
+type TestContext struct {
```

Comment:
SGTM! But different components have different `TestContext`, such as:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/garbagecollector/garbage_collector_test.go#L200-L210
```go
type testContext struct {
	tearDown           func()
	gc                 *garbagecollector.GarbageCollector
	clientSet          clientset.Interface
	apiExtensionClient apiextensionsclientset.Interface
	dynamicClient      dynamic.Interface
	metadataClient     metadata.Interface
	startGC            func(workers int)
	// syncPeriod is how often the GC started with startGC will be resynced.
	syncPeriod time.Duration
}
```
https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/admissionwebhook/admission_test.go#L69-L80

```go
type testContext struct {
	t *testing.T

	admissionHolder *holder

	client    dynamic.Interface
	clientset clientset.Interface
	verb      string
	gvr       schema.GroupVersionResource
	resource  metav1.APIResource
	resources map[schema.GroupVersionResource]metav1.APIResource
}
```
https://github.com/kubernetes/kubernetes/blob/master/test/integration/volumescheduling/util.go#L42-L52

```go
type testContext struct {
	closeFn         framework.CloseFunc
	httpServer      *httptest.Server
	ns              *v1.Namespace
	clientSet       *clientset.Clientset
	informerFactory informers.SharedInformerFactory
	scheduler       *scheduler.Scheduler

	ctx      context.Context
	cancelFn context.CancelFunc
}
```

These `testContext` are very different. I'm a little concerned about whether it's appropriate to make `TestContext` as a public dependency.

Reply:
Actually, I am moving the util functions of sig-scheduling and sig-node integration test. For different definitions of `testContext`, we can permit different sig has their own definition. If they decide to move it into the util. they can adjust the definition of `TestContext` on util to adapt.

This PR is just for adapting changes of moving TainBasedEviction test from sig-scheduling to sig-node. https://github.com/kubernetes/kubernetes/issues/87427